image/tools/lib/component/mysql.sh

```diff
@@ -1,8 +1,13 @@
-function component_get_url {
-    echo 'mysql://localhost'
-}
-
 function component_dump_data {
-    auth_url=$1
-    echo "Use $auth_url to archive mysql data"
-}
+    dest=$1
+    databases=$(mysql -h$MYSQL_HOST -u$MYSQL_USER  -p$MYSQL_PASSWORD -e 'SHOW DATABASES' | tail -n+2 | grep -v information_schema)
+    for database in $databases; do
+        ts=$(date '+%H:%M:%S')
+        mysqldump -h$MYSQL_HOST -u$MYSQL_USER -p$MYSQL_PASSWORD -R $database | gzip > $dest/archives/$database-$ts.dump.gz
```

Comment:
There might be multiple mysql instances, should the archive name have the namespace or some identifier attached to it so that we can see where it was coming from?

Reply:
I designed it for 1 job per component/database.

Will there be a case where we need the same job to do backup different dbs?

Reply:
Not sure if that's the case with mysql,  but postgres is used at least in enmasse, codeready and sso.

Reply:
But you're right, they will be handled by different scripts, not the same script backing up the databases of those services.